File: api/src/main/java/sh/foxboy/bapp/api/punishment/PunishmentManager.java
package sh.foxboy.bapp.api.punishment;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import sh.foxboy.bapp.api.entity.Arbiter;
import sh.foxboy.bapp.api.entity.User;

import java.util.Date;
import java.util.List;

public interface PunishmentManager {

	@NotNull
	public List<Punishment> getPunishments();

	@NotNull
	public List<Punishment> getPunishments(@NotNull SortBy order);

	@NotNull
	public List<Punishment> getPunishments(@NotNull SortBy order, @NotNull Integer page);

	@NotNull
	public List<Punishment> getPunishments(@NotNull SortBy order, @NotNull Integer page, @NotNull Integer pageSize);

	@NotNull
	public List<Punishment> getPunishments(@NotNull Integer page);

	@NotNull
	public List<Punishment> getPunishments(@NotNull Integer page, @NotNull Integer pageSize);

	@NotNull
	public Punishment createPunishment(@NotNull PunishmentType type, @NotNull Arbiter arbiter, @Nullable User target, @NotNull String reason, @NotNull Date expiry);

	@Nullable
	public Punishment deletePunishment(@NotNull Punishment punishment);
}

File: bukkit/src/main/kotlin/punishment/PunishmentManager.kt
package sh.foxboy.bapp.punishment

import org.bukkit.OfflinePlayer
import org.jetbrains.exposed.sql.selectAll
import sh.foxboy.bapp.WithPlugin
import sh.foxboy.bapp.api.punishment.Punishment
import sh.foxboy.bapp.api.punishment.PunishmentManager
import sh.foxboy.bapp.api.punishment.PunishmentType
import sh.foxboy.bapp.api.punishment.SortBy
import sh.foxboy.bapp.database.tables.PunishmentsTable
import java.util.Date

class PunishmentManager : PunishmentManager, WithPlugin {
    override fun getPunishments(): List<Punishment> {
        return getPunishments(SortBy.DATE_ASC)
    }

    override fun getPunishments(sortBy: SortBy): List<Punishment> {
        return getPunishments(sortBy, 1)
    }

    override fun getPunishments(sortBy: SortBy, page: Int): List<Punishment> {
        return getPunishments(sortBy, 1, 8)
    }

    override fun getPunishments(order: SortBy, page: Int, pageSize: Int): List<Punishment> {
        return this.plugin.postgresHandler.getPunishments(PunishmentsTable.selectAll(), order, page, pageSize)
    }

    override fun getPunishments(page: Int): List<Punishment> {
        return getPunishments(SortBy.DATE_ASC, page, 8)
    }

    override fun getPunishments(page: Int, pageSize: Int): List<Punishment> {
        return getPunishments(SortBy.DATE_ASC, page, pageSize)
    }

    override fun createPunishment(
        type: PunishmentType,
        arbiter: OfflinePlayer,
        target: OfflinePlayer?,
        reason: String,
        expiry: Date
    ): Punishment {
        return Punishment(type, arbiter, target, reason, expiry)
    }

    override fun deletePunishment(punishment: Punishment): Punishment {
        TODO("Not yet implemented")
    }
}